packages/shared/hooks/useRefAutoFocus/useRefAutoFocus.ts

```diff
@@ -0,0 +1,20 @@
+import { DependencyList, MutableRefObject, useEffect, useRef } from 'react';
+
+/**
+ * Returns `ref` object that is automatically focused when `canFocus` is `true`.
+ * Focus can be also re triggered by changing any of the `refocusDeps`.
+ */
+export function useRefAutoFocus<T extends { focus(): void }>(options: {
+  canFocus: boolean;
+  refocusDeps?: DependencyList;
+}): MutableRefObject<T> {
+  const ref = useRef<T>();
+
+  useEffect(() => {
+    if (options.canFocus) {
```

Comment:
Linguistic nit: An option name such as `canFocus` makes sense when talking from the overall perspective of the problem you were trying to fix with this PR: a `<Document>` cannot be focused when it's not visible so we should not attempt to call `focus` in that scenario.

However, from the perspective of this hook in separation, `useRefAutoFocus` always _can_ focus a ref, it's just a matter of whether it _should_ or _shouldn't_ do that, isn't it?

Reply:
Absolutely! I changed the name to `shouldFocus`.